### PR TITLE
Add the ability to change chroma implementation

### PR DESCRIPTION
**Pull request by charliems**
# Description

Added the ability to use a local chroma store instead of server by changing `chroma_api_impl="duckdb+parquet"` in `ChromaReader`

## Type of Change

- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

**Comment by jeffchuber** [comment]
@charliems thanks for adding this!

I think you want to change `chroma_db_impl` however? 

```
from chromadb.config import Settings
client = chromadb.Client(Settings(
    chroma_db_impl="duckdb+parquet",
    persist_directory="/path/to/persist/directory" # Optional, defaults to .chromadb/ in the current directory
))
```

**Comment by charliems** [comment]
Yes I do. By default ChromaReader tries to use API on localhost:8000 so if you only have a folder with no server, you get a request error as it tries reaching the endpoint. To fix this would you just set `chroma_api_impl=None` or would you also have to have `chroma_db_impl=duckdb+parquet`?

**Comment by jeffchuber** [comment]
@charliems i think both.

@jerryjliu suggestions here?

File: llama_index/readers/chroma.py [hunk 1]
```diff
@@ -39,7 +41,8 @@ def __init__(
 
         self._client = chromadb.Client(
             Settings(
-                chroma_api_impl="rest",
+                chroma_api_impl=chroma_api_impl,
+                chroma_db_impl=chroma_db_impl or "chromadb.db.duckdb.DuckDB",
```
Comment (jerryjliu) [inline llama_index/readers/chroma.py:45]: is "chromadb.db.duckdb.DuckDB" the default string value?

**Review by jerryjliu — APPROVED** [review]
yeah i think you'd just set `chroma_api_impl=None` and `chroma_db_impl="duckdb+parquet"`? following guide here: https://docs.trychroma.com/api-reference

thanks for the change

File: llama_index/readers/chroma.py (hunk 1 above)
Comment (charliems) [reply llama_index/readers/chroma.py:45]: I believe so
https://github.com/chroma-core/chroma/blob/main/chromadb/config.py#L42

**Pull request merged**